### PR TITLE
BIP68/113 for generation transaction

### PR DESCRIPTION
**Pull request by jl2012**
To clarify the applicability of BIP68 and 113 to generation transaction

### CODE REVIEW

**Comment by luke-jr** [comment]
@maaku @btcdrak @NicolasDorier @afk11

**Comment by maaku** [comment]
This isn't true. The rules apply as much to outputs of a coinbase as they do any other transaction. Is the proposed text talking about the coinbase -1 nSequence value? Then that should be specified exactly, although I don't think the clarification is necessary (coinbase is already specially treated).

**Comment by jl2012** [comment]
@maaku, isn't the SequenceLocks is done only for non-coinbase tx?

https://github.com/bitcoin/bitcoin/blob/master/src/main.cpp#L2480

(I'm talking about the BIP68, not BIP112)

**Comment by luke-jr** [comment]
nSequence is about inputs, not outputs... and the new rule for it doesn't make sense for the generation tx input since it's not spending coins.

**Comment by afk11** [comment]
For 113 (MTP), the generation transaction's locktime is checked, so looks good to me: https://github.com/bitcoin/bitcoin/blob/master/src/main.cpp#L3602

@jl2012 Maybe clarify by saying 68 doesn't apply to the inputs of the coinbase transaction? It can certainly apply to it's outputs.

**Comment by jl2012** [comment]
@maaku @afk11 @luke-jr  Edited

File: bip-0068.mediawiki [hunk 1]
```diff
@@ -49,6 +49,8 @@ The block produced time is equal to the median-time-past of its previous block.
 
 When the relative lock-time is block-based, it is interpreted as a minimum block-height constraint over the input's age. A relative block-based lock-time of zero indicates an input which can be included in any block. More generally, a relative block lock-time n can be included n blocks after the mining date of the output it is spending, or any block thereafter.
 
+The new rules are not applied to the nSequence field of the input of generation transaction.
```
Comment (maaku) [inline bip-0068.mediawiki]: "of the generation"

Is this the proper terminology? Should we say "coinbase transaction" here? Honestly not sure what the current accepted terminology is.

**Comment by maaku** [comment]
Right, the sequence lock checks are not done for the coinbase input. The updated text is better (see comment made)

File: bip-0068.mediawiki (hunk 1 above)
Comment (btcdrak) [inline bip-0068.mediawiki]: "coinbase tx" is definitely more common.

**Comment by jl2012** [comment]
@maaku @btcdrak fixed

**Comment by afk11** [comment]
ACK 8d40b6e

**Comment by btcdrak** [comment]
ACK 8d40b6e

**Comment by maaku** [comment]
ACK 8d40b6e

File: bip-0068.mediawiki (hunk 1 above)
Comment (luke-jr) [inline bip-0068.mediawiki]: The coinbase is the input scriptSig for the generation transaction.

**Comment by luke-jr** [comment]
Disagree with "coinbase transaction", but this is hardly the first time, and an ACK is an ACK...

**Pull request merged**